lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -130,8 +130,25 @@ public void setScorer(Scorable scorer) throws IOException {
         public void collect(int doc) throws IOException {
           ++totalHits;
           hitsThresholdChecker.incrementHitCount();
-          if (queueFull) {
-            if (collectedAllCompetitiveHits || reverseMul * comparator.compareBottom(doc) <= 0) {
+
+          Object currentValue = null;
+
+          if (fieldValueChecker != null) {
+            try {
+              currentValue = comparator.getDocValue(doc);
```

Comment:
We should consider this operation as very costly. Getting the actual value of a string field means resolving the ordinal and we have field comparator that can avoid this lookup efficiently.

Reply:
I agree and disagree -- some comparators can be costly to get the leaf values from, as the comment highlights. However, for the remaining comparators (Int, Float, Double, Long), this is a fairly cheap operation